### ISSUE TITLE
Arithmetic operations with None not detected without strict-optional

### ISSUE THREAD

**Issue by pkch** [python/mypy#3121]
(Based on #3111)

Without `--strict-optional`, some invalid `None` opertations are undetected:

```
None + 1 # ok
None * 1 # ok
None / 1 # ok
None - 1 # ok

None + '' # error
None * '' # ok
None / '' # error
None - '' # error
```

With `--strict-optional`, all of them fail type check but with inconsistent error messages:

```
Unsupported operand types for + (None and "int")
Unsupported operand types for * (None and "int")
Unsupported operand types for / (None and "int")
Unsupported operand types for - (None and "int")
Unsupported left operand type for + (None)
Unsupported operand types for * (None and "str")
Unsupported left operand type for / (None)
Unsupported left operand type for - (None)
```

**Comment by ilevkivskyi** [comment]
The plan is to make ``--strict-optional`` turned on by default, so that the first part does not need to be fixed. However, the error messages could be more homogeneous indeed.

**Comment by pkch** [comment]
I guess we should fix a few of tests then because everyone likes to put `None + 1` as an example of what should fail (like in module import test, etc.). Or are we going to start running tests with `--strict-optional` in the next few days?

**Comment by ilevkivskyi** [comment]
> Or are we going to start running tests with ``--strict-optional`` in the next few days?

No, this is rather a long-term goal. However, there are dedicated tests for this flag: ``check-optional.test``

**Comment by msullivan** [comment]
Strict optional is the default now, so I'm going to say this is "working as intended"
